benchmark-service/test/test_update_commits.py

```diff
@@ -0,0 +1,55 @@
+import json
+import logging
+import asyncio
+import pytest
+import aiohttp
+
+from hailtop.config import get_deploy_config
+from hailtop.auth import service_auth_headers
+from hailtop.tls import in_cluster_ssl_client_session, get_context_specific_ssl_client_session
+import hailtop.utils as utils
+
+pytestmark = pytest.mark.asyncio
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+sha = 'd626f793ad700c45a878d192652a0378818bbd8b'
+commit = None
+
+
+async def test_update_commits():
+    deploy_config = get_deploy_config()
+    headers = service_auth_headers(deploy_config, 'benchmark')
+    commit_benchmark_url = deploy_config.url('benchmark', f'/api/v1alpha/benchmark/commit/{sha}')
+
+    async with get_context_specific_ssl_client_session(
+            raise_for_status=True,
+            timeout=aiohttp.ClientTimeout(total=60)) as session:
+
+        global commit
```

Comment:
Let's define commit = None here instead of above. Then you don't need this to be global here.

Reply:
but then it says commit is referenced before defining in the while loop

Reply:
ah unless i use 'nonlocal'